esphome/components/rx8130/rx8130.cpp

```diff
@@ -0,0 +1,125 @@
+#include "rx8130.h"
+#include "esphome/core/log.h"
+
+// https://download.epsondevice.com/td/pdf/app/RX8130CE_en.pdf
+
+namespace esphome {
+namespace rx8130 {
+using namespace i2c;
+
+static const char *const TAG = "rx8130";
+
+void RX8130Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up RX8130...");
+  uint8_t clear_flags = 0x00;
+  // Set digital offset to disabled with no offset
+  if (this->write_register(RX8130_REG_DIG_OFFSET, &clear_flags, 1) != ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  // Disable wakeup timers
+  if (this->write_register(RX8130_REG_EXTEN, &clear_flags, 1) != ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  // Clear VLF flag in case there has been data loss
+  if (this->write_register(RX8130_REG_FLAG, &clear_flags, 1) != ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  // Clear test flag and disable interrupts
+  if (this->write_register(RX8130_REG_CTRL0, &clear_flags, 1) != ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  // Enable battery charging and switching
+  uint8_t battery_flags = 0x30;
+  if (this->write_register(RX8130_REG_CTRL1, &battery_flags, 1) != ERROR_OK) {
+    this->mark_failed();
+    return;
+  }
+  // Clear STOP bit
+  this->stop_(false);
+}
+
+void RX8130Component::update() { this->read_time(); }
+
+void RX8130Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "RX8130:");
+  LOG_I2C_DEVICE(this);
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "Communication with RX8130 failed!");
+    return;
+  }
+  ESP_LOGCONFIG(TAG, "  DOFFS  %s", format_bin((uint8_t) this->reg(RX8130_REG_DIG_OFFSET)).c_str());
+  ESP_LOGCONFIG(TAG, "  EXTEN  %s", format_bin((uint8_t) this->reg(RX8130_REG_EXTEN)).c_str());
+  ESP_LOGCONFIG(TAG, "  FLAGS  %s", format_bin((uint8_t) this->reg(RX8130_REG_FLAG)).c_str());
+  ESP_LOGCONFIG(TAG, "  CTRL0  %s", format_bin((uint8_t) this->reg(RX8130_REG_CTRL0)).c_str());
+  ESP_LOGCONFIG(TAG, "  CTRL1  %s", format_bin((uint8_t) this->reg(RX8130_REG_CTRL1)).c_str());
+  ESP_LOGCONFIG(TAG, "  Timezone: '%s'", this->timezone_.c_str());
+}
+
+float RX8130Component::get_setup_priority() const { return setup_priority::DATA; }
+
+void RX8130Component::read_time() {
+  uint8_t date[7];
+  if (this->read_register(RX8130_REG_SEC, date, 7) != ERROR_OK) {
+    ESP_LOGE(TAG, "Can't read I2C data.");
+    this->mark_failed();
```

Comment:
We shouldn't fail the component if we failed to read once. Maybe use a `set_retry` to try multiple times before actually failing it?
```suggestion
    this->status_set_warning(ESP_LOG_MSG_COMM_FAIL);
```

Reply:
Removed